File: WebShop/WebShop-Client/src/app/_models/customer/DirectCustomerResponse.ts
import { StaticAccountResponse } from "../account";
import { StaticCountryResponse } from "../country";
import { StaticOrderResponse } from "../order";
import { StaticPaymentResponse } from "../payment";

export interface DirectCustomerResponse
{
	customerID: number;
	account: StaticAccountResponse;
	firstName: string;
	lastName: string;
	phoneNumber: string;
	country: StaticCountryResponse;
	zipCode: number;
	gender: string;
	created_At: Date;
	payments: StaticPaymentResponse[];
	orders: StaticOrderResponse[];
}

File: WebShop/WebShop-API/DTOs/Customer/DirectCustomerResponse.cs
namespace WebShop_API.DTOs.Customer
{
    public class DirectCustomerResponse
    {
        public int CustomerID { get; set; }

        public StaticAccountResponse Account { get; set; } = null!;

        public string FirstName { get; set; } = string.Empty;

        public string LastName { get; set; } = string.Empty;

        public string PhoneNumber { get; set; } = string.Empty;

        public StaticCountryResponse Country { get; set; } = null!;

        public int ZipCode { get; set; } = 0;

        public string Gender { get; set; } = string.Empty;

        public DateTime Created_At { get; set; }

        public List<StaticPaymentResponse> Payments { get; set; } = new List<StaticPaymentResponse>();

        public List<StaticOrderResponse> Orders { get; set; } = new List<StaticOrderResponse>();
    }
}
